libexec/mdl-install-plugin.sh

```diff
@@ -0,0 +1,163 @@
+      \?) echo "${red}Invalid option: -$OPT$norm" >&2 ;;
+      *) echo "${red}Some of these options are invalid:$norm $*" >&2; exit 2 ;;
+   esac
+done
+shift $((OPTIND - 1))
+
+requires basename curl find jq grep head mktemp unzip xargs "${MDL_CONTAINER_TOOL[0]}"
+
+# Calculate the plugin path based on type
+function plugin_type_path() {
+   case $1 in
+      antivirus) echo lib/antivirus ;;
+      assignsubmission) echo mod/assign/submission ;;
+      assignfeedback) echo mod/assign/feedback ;;
+      booktool) echo mod/book/tool ;;
+      customfield) echo customfield/field ;;
+      datafield) echo mod/data/field ;;
+      datapreset) echo mod/data/preset ;;
+      ltisource) echo mod/lti/source ;;
+      fileconverter) echo files/converter ;;
+      ltiservice) echo mod/lti/service ;;
+      mlbackend) echo lib/mlbackend ;;
+      forumreport) echo mod/forum/report ;;
+      quiz) echo mod/quiz/report ;;
+      quizaccess) echo mod/quiz/accessrule ;;
+      scormreport) echo mod/scorm/report ;;
+      workshopform) echo mod/workshop/form ;;
+      workshopallocation) echo mod/workshop/allocation ;;
+      workshopeval) echo mod/workshop/eval ;;
+      block) echo blocks ;;
+      qtype) echo question/type ;;
+      qbehaviour) echo question/behaviour ;;
+      qformat) echo question/format ;;
+      editor) echo lib/editor ;;
+      atto) echo lib/editor/atto/plugins ;;
+      tool) echo admin/tool ;;
+      logstore) echo admin/tool/log/store ;;
+      availability) echo availability/condition ;;
+      calendartype) echo calendar/type ;;
+      message) echo message/output ;;
+      format) echo course/format ;;
+      dataformat) echo dataformat ;;
+      profilefield) echo user/profile/field ;;
+      coursereport) echo course/report ;;
+      gradeexport) echo grade/export ;;
+      gradeimport) echo grade/import ;;
+      gradereport) echo grade/report ;;
+      gradingform) echo grade/grading/form ;;
+      mnetservice) echo mnet/service ;;
+      search) echo search/engine ;;
+      media) echo media/player ;;
+      plagiarism) echo plagiarism ;;
+      cachestore) echo cache/stores ;;
+      cachelock) echo cache/locks ;;
+      contenttype) echo contentbank/contenttype ;;
+      h5plib) echo h5p/h5plib ;;
+      qbank) echo question/bank ;;
+      *) echo "$1" ;;
+   esac
+}
+
+for mname in $mnames; do
+
+   # Get the Moodle container for this environment
+   container="$(container_tool ps -f "label=com.docker.compose.project=$mname" --format '{{.Names}}' | grep moodle | head -1)"
+   [[ -z $container ]] && echo "${red}Could not find a container running Moodle for $ul$mname$rmul!$norm" >&2 && exit 1
+
+   # Get Moodle base directory by inspecting the container mount
+   base_dir=$(container_tool inspect "$container" | jq -r '.[] .Mounts[] | select(.Name != null and (.Name | contains("src"))) | .Destination')
+   [[ -z $base_dir ]] && echo "${red}Could not determine Moodle base directory for $ul$mname$rmul!$norm" >&2 && exit 1
```

Comment:
FYI, if you see this and wonder where this is coming from, this is a new approach as of the Bitnami impact.  Whereas everything works right now since we're still pointing to Bitnami images, we have some brittle code that hard-codes `/bitnami/*` paths.  This new approach is very safe because it literally queries the container tool for the mounted path of the src volume based on its configuration inspection tooling. See #13.